includes/class-wc-countries.php

```diff
@@ -57,6 +57,24 @@ public function get_countries() {
 		return $this->countries;
 	}
 
+	/**
+	 * Check if a given code represents a valid ISO 3166-1 alpha-2 code for a country known to us.
+	 *
+	 * @since 5.1.0
+	 * @param string $country_code The country code to check as a ISO 3166-1 alpha-2 code.
+	 * @param bool   $throw True to throw an exception if the country code isn't recognized.
+	 * @return bool True if the country is known to us, false otherwise.
+	 * @throws Exception The country code isn't recognized, and the $throw argument was passed as true.
+	 */
+	public function country_exists( $country_code, $throw = false ) {
+		$exists = array_key_exists( $country_code, $this->get_countries() );
+		if ( ! $exists && $throw ) {
+			/* translators: %s: ISO 3166-1 alpha-2 country code. */
+			throw new Exception( sprintf( __( "'%s' is not a valid country code", 'woocommerce' ), $country_code ) );
```

Comment:
what do you think about moving the exception throwing logic to the parent code, to make the code there more directly readable?

Reply:
I was trying to provide a centralized point for exceptioning, but sure, readabilty is important. I've done the change.